pkg/query/rules.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package query
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+)
+
+func NewRulesRetriever(rs storepb.RulesServer) *rulesRetriever {
+	return &rulesRetriever{
+		proxy: rs,
+	}
+}
+
+type rulesRetriever struct {
+	proxy storepb.RulesServer
+}
+
+func (rr *rulesRetriever) RuleGroups(ctx context.Context) ([]*storepb.RuleGroup, storage.Warnings, error) {
+	resp := &rulesServer{ctx: ctx}
+
+	if err := rr.proxy.Rules(&storepb.RulesRequest{
+		PartialResponseStrategy: storepb.PartialResponseStrategy_ABORT,
```

Comment:
This should be configureable, no?

Reply:
yes :+1: I was just impatient and wanted to have something working :)

Reply:
plus (note to myself): once I have also configurability injected here, this is where I want to implement the deduplication, and of course it needs unit tests here too.